src/nifake/unit_tests/test_grpc.py

```diff
@@ -163,6 +166,40 @@ def _assert_call(self, function_name, request_object):
 
     # Methods
 
+    def test_new_session_already_exists(self):
+        library_func = 'InitWithOptions'
+        session_name = 'existing_session'
+        error_message = "Cannot initialize '" + session_name + "' when a session already exists."
+        grpc_error = grpc.StatusCode.ALREADY_EXISTS
+        self._set_side_effect(library_func, side_effect=MyRpcError(None, error_message, grpc_error=grpc_error))
+        init_behavior = nifake.SessionInitializationBehavior.INITIALIZE_SERVER_SESSION
+        grpc_options = nifake.GrpcSessionOptions(object(), session_name, init_behavior)
+        interpreter = nifake._grpc_stub_interpreter.GrpcStubInterpreter(grpc_options)
+        try:
+            interpreter.init_with_options('dev1', False, False, '')
+            assert False
+        except nifake.Error as e:
+            assert e.rpc_code == grpc_error
+            assert e.description == error_message
+            assert str(e) == f'StatusCode.ALREADY_EXISTS: {error_message}'
```

Comment:
assert e.rpc_code == grpc_error
Minor thing: you could use `{grpc_error}` here, since this is an fstring. That would also be consistent with your change to test_system_nidmm.py.

Reply:
This applies to test_attach_to_non_existent_session, as well.

Reply:
I did this intentionally, to 1) show you what it'll actually look like, and 2) assert that it stringifies in the way we expect, and not just as "1" or "2", etc.